### DISCUSSION TITLE
Hub menu indexing to the next program on spike prime

### DISCUSSION THREAD

**Discussion by sairampt** [pybricks/support#1910] [Q&A]
Hi. I tried the hub_menu to load multiple programs on the spike prime hub and named it like 1,2, 3 etc and it works fine. Is there a way to keep the program number active on the hub or move to the next program without having to run the program again, pick the program and then run?

For example, I run the main hub_menu program by pressing the center button. The display shows 1 and then pressing the center button runs the program tied to 1. After the program is completed, is there a way to show 2 and just pressing the center button will run the program associated with 2 etc.

**Comment by dlech** [comment]
You can use `hub.system.storage()` to save small data like this between program runs. 

https://docs.pybricks.com/en/stable/hubs/primehub.html#pybricks.hubs.PrimeHub.system.storage

**Comment by kkhanmd** [comment]
I have been trying to ask the same here as well
https://github.com/orgs/pybricks/discussions/1878
